packages/ui/src/components/Modal/Dialog.tsx

```diff
@@ -0,0 +1,155 @@
+
+const StyledBackdrop = styled.div<{ 'data-open': boolean }>`
+  position: fixed;
+  top: 0;
+  right: 0;
+  height: 0;
+  width: 0;
+  overflow: hidden;
+  background-color: ${({ theme }) => theme.colors.overlay};
+
+  &[data-open='true'] {
+    padding: ${({ theme }) => theme.space['2']};
+    overflow: auto;
+    display: flex;
+    bottom: 0;
+    left: 0;
+    height: 100%;
+    width: 100%;
+  }
+`
+
+type StyledDialogProps = {
+  'data-size': ModalSize
+  'data-placement': ModalPlacement
+}
+
+const StyledDialog = styled.dialog<StyledDialogProps>`
+  background-color: ${({ theme }) =>
+    theme.colors.neutral.backgroundWeakElevated};
+  position: relative;
+  border-radius: ${({ theme }) => theme.radii.default};
+  border: 0;
+  padding: ${({ theme }) => theme.space['3']};
+  width: ${MODAL_WIDTH.medium}px;
+  box-shadow: ${({ theme }) => theme.shadows.modal};
+
+  ${Object.entries(MODAL_WIDTH).map(
+    ([size, value]) => `
+      &[data-size="${size}"] {
+        width: ${value}px;
+      }
+      `,
+  )}
+
+  ${Object.entries(MODAL_PLACEMENT).map(
+    ([placement, value]) => `
+        &[data-placement="${placement}"] {
+          ${value}
+        }
+        `,
+  )}
+`
+
+export const Dialog = ({
+  children,
+  open,
+  placement,
+  onClose,
+  hideOnClickOutside,
+  size,
+  id,
+  ariaLabel,
+  className,
+  'data-testid': dataTestId,
+  preventBodyScroll,
+  hideOnEsc,
+  backdropClassName,
+  dialogCss,
+  backdropCss,
+}: DialogProps) => {
+  const containerRef = useRef(document.createElement('div'))
+  const onCloseRef = useRef(onClose)
+
+  // Portal to put the modal in
+  useEffect(() => {
+    const element = containerRef.current
+    document.body.appendChild(element)
+
+    return () => {
+      document.body.removeChild(element)
+    }
+  }, [])
+
+  // Save the reassignment of eventHandler in the useEffect below
+  useEffect(() => {
+    onCloseRef.current = onClose
+  }, [onClose])
+
+  // Handle hide on esc press
+  useEffect(() => {
+    const handleEscPress = (event: KeyboardEvent) => {
+      if (event.key === 'Escape' && hideOnEsc) {
+        onCloseRef.current()
+      }
+    }
+    if (open) {
+      document.addEventListener('keyup', handleEscPress)
+    }
+
+    return () => {
+      document.removeEventListener('keyup', handleEscPress)
+    }
+  }, [open, onCloseRef, hideOnEsc])
+
+  // Handle body scroll
+  useEffect(() => {
+    if (open && preventBodyScroll) {
+      document.body.style.overflow = 'hidden'
+    } else {
+      document.body.style.overflow = 'auto'
+    }
+  }, [preventBodyScroll, open])
+
+  // Stop click to prevent unexpected dialog close
+  const stopClick: MouseEventHandler = useCallback(event => {
+    event.stopPropagation()
+  }, [])
+
+  // Stop key up : used when having inputs in modals
+  const stopKeyUp: KeyboardEventHandler = useCallback(event => {
+    event.stopPropagation()
+  }, [])
+
+  return createPortal(
```

Comment:
The create portal should be dynamic when the user open the modal or the dom will be full of useless div. You can refer to Popup internal component which is used by tooltip

![Screenshot 2023-08-30 at 11 37 46](https://github.com/scaleway/ultraviolet/assets/15812968/497251f8-40e9-4ff3-9d7b-8bfab8c43210)